crates/rooch-framework/sources/nft/collection.move

```diff
@@ -0,0 +1,291 @@
+        let mutator_ref = context::new_object_with_owner(
+            ctx,
+            object_ref::owner(collection),
+            MutatorRef {
+                collection: object_ref::id(collection),
+            }
+        );
+        mutator_ref
+    }
+
+    public fun destroy_mutator_ref(mutator_ref :ObjectRef<MutatorRef>):ObjectID{
+        assert_mutator_exist_of_ref(&mutator_ref);
+        let MutatorRef {
+            collection
+        } = object_ref::remove<MutatorRef>(mutator_ref);
+        collection
+    }
+
+    public fun get_collection_id(mutator: &ObjectRef<MutatorRef>): ObjectID{
+        assert_mutator_exist_of_ref(mutator);
+        let mutator_object_ref = object_ref::borrow(mutator);
+        mutator_object_ref.collection
+    }
+
+
+    public(friend) fun increment_supply(mutator: &ObjectRef<MutatorRef>, ctx: &mut Context): Option<u64>{
+        assert_mutator_exist_of_ref(mutator);
+        let mutator_object_ref = object_ref::borrow(mutator);
+        assert_collection_exist_of_id(mutator_object_ref.collection, ctx);
+        let collection_object_mut_ref = context::borrow_object_mut<Collection>(ctx, mutator_object_ref.collection);
+        let collection_mut_ref = object::borrow_mut(collection_object_mut_ref);
+        collection_mut_ref.supply.current = collection_mut_ref.supply.current + 1;
+        if(option::is_some(&collection_mut_ref.supply.maximum)){
+            assert!(collection_mut_ref.supply.current <= *option::borrow(&collection_mut_ref.supply.maximum), ErrorCollectionMaximumSupply);
+            option::some(collection_mut_ref.supply.current)
+        }else{
+            option::none<u64>()
+        }
+    }
+
+    public (friend) fun decrement_supply(mutator: &ObjectRef<MutatorRef>, ctx: &mut Context): Option<u64>{
+        assert_mutator_exist_of_ref(mutator);
+        let mutator_object_ref = object_ref::borrow(mutator);
+        assert_collection_exist_of_id(mutator_object_ref.collection, ctx);
+        let collection_object_mut_ref = context::borrow_object_mut<Collection>(ctx, mutator_object_ref.collection);
+        let collection_mut_ref = object::borrow_mut(collection_object_mut_ref);
+        collection_mut_ref.supply.current = collection_mut_ref.supply.current - 1;
+        if(option::is_some(&collection_mut_ref.supply.maximum)){
+            option::some(collection_mut_ref.supply.current)
+        }else{
+            option::none<u64>()
+        }
+    }
+
+    // assert
+    public fun assert_collection_exist_of_ref(collectionRef: &ObjectRef<Collection>){
+        assert!( object_ref::exist_object(collectionRef), ErrorCollectionNotExist);
+    }
+
+    public fun assert_collection_exist_of_id(collectionID: ObjectID, ctx: & Context){
+        assert!( context::exist_object(ctx, collectionID), ErrorCollectionNotExist);
+        context::borrow_object<Collection>(ctx,collectionID);
```

Comment:
> Unused borrow

Because it is necessary to determine that the Object corresponding to the Object ID is the current type

This should be refactored after the subsequent Object module upgrade.